CMakeLists.txt

```diff
@@ -307,6 +307,12 @@ else()
     find_package(ZSTD)
 endif()
 
+#
+# ZLIB
+#
+set(ZLIB_FIND_REQUIRED ON)
+find_package(ZLIB)
```

Comment:
It looks as a reconstruction of 4bd44330157819c85ece060c1186cff41bae1ee1 logic, but a bit differently. I would rather backport the original commit: it provides our own FindZLIB.cmake (so it will not be changed by a cmake update -- AFAIK we do that for all Find\*.cmake files) and uses the same paths everywhere across the build scripts.

In such cases I rewrite the original commit message with an explanation why it is backported, see d8460466680f07f7cd75c4a1b273f20a18a2f3a5 for example.

Please, place the fix before the BuildLibCURL.cmake update: it is desirable to keep every commit working to make `git bisect` easier.